### ISSUE TITLE
Does not scan already scanned files

### ISSUE THREAD

**Issue by ericdallo** [clj-kondo/clj-kondo#1395]
**version**

2021.09.25

**platform**

JVM

**editor**

Tested on both clojure-lsp, calva and cli clj-kondo

**problem**

If the `--lint` receives a folder and a file inside that same folder, it seems that clj-kondo scan the same twice.

**repro**

Repro project: https://github.com/paulbutcher/heisenwarning

```bash
$ clj -Sdeps '{:deps {clj-kondo/clj-kondo {:mvn/version "2021.09.25"}}}' -m clj-kondo.main --lint test test/clj/race_and_improve/session_test.clj 
test/clj/race_and_improve/session_test.clj:4:1: warning: redefined var #'race-and-improve.session-test/simple-test
linting took 187ms, errors: 0, warnings: 1
```

**config**

No config

**expected behavior**

Expect to kondo scan only a single time or at least doesn't report false-positves

**Comment by borkdude** [comment]
OK, we can make an atom with already linted file names. Also we can deduplicate directories and the classpath before even inspecting the sources.

**Comment by borkdude** [comment]
Why would anyone call clj-kondo like this btw? Why not just call it with only the `test` dir?

**Comment by borkdude** [comment]
Any follow up on this?

**Comment by ericdallo** [comment]
Sorry for the confusion, 10 days and I was not remembering why that was happening 😅
I found the slack thread that caused this issue, [here](https://clojurians.slack.com/archives/CBE668G4R/p1633190003410400?thread_ts=1632674357.358200&cid=CBE668G4R) you can see that the the repro project has a `project.clj`and a `deps.edn`, the former has `test` on the classpath and the latter has `test/clj`, making kondo analyze the file `test/clj/race_and_improve/session_test.clj` two times.

Does that makes sense?

**Comment by borkdude** [comment]
@ericdallo ok, another attempt, pushed to the branch.

**Comment by ericdallo** [comment]
I found an easier repro that may help you @borkdude, using the clj-kondo repo itself, running from clj-kondo repo:
```
$ clj -m clj-kondo.main --lint test test/clj_kondo/format_test.clj 
test/clj_kondo/format_test.clj:7:1: warning: redefined var #'clj-kondo.format-test/format-test
linting took 2275ms, errors: 0, warnings: 1
```

**Comment by ericdallo** [comment]
Also, I created a test reproducing the issue [here](https://github.com/ericdallo/clj-kondo/commit/bfeb73ec91415d37fb4cabf329cfd822bc5d8dec)
